cloudnet/src/main/java/de/dytanic/cloudnet/console/animation/questionlist/answer/QuestionAnswerTypeHostAndPort.java

```diff
@@ -24,11 +35,14 @@ public boolean isValidInput(@NotNull String input) {
             URI uri = URI.create("tcp://" + input);
 
             String host = uri.getHost();
-            if (host == null) {
+            if (host == null || (this.requiresPort && uri.getPort() == -1)) {
                 return null;
             }
 
             InetAddress inetAddress = InetAddresses.forUriString(host);
+            if (this.requiresPort && !PortValidator.checkHost(inetAddress.getHostAddress(), uri.getPort())) {
```

Comment:
I think we should check the host too if the port is not required. This might need a new method in the PortValidator - when address already in use or binding success return true otherwise return false.